### PR TITLE
DRILL-7795: Add support for overriding hostname to C++ client

### PR DESCRIPTION
**Pull request by jduo**
Also set the TLS SNI property to the host field if the hostname wasn't overriddeen.

# [DRILL-7795](https://issues.apache.org/jira/browse/DRILL-7795): Add support for overriding hostname to C++ client

## Description
Add the property overrideHostname to the C++ client. If this property is not set, still set the TLS SNI property to the hostname being connected to.

## Documentation
The C++ client now sees 

## Testing
Using the ODBC Driver, with this custom client substituted in.

### CODE REVIEW

File: contrib/native/client/src/clientlib/channel.cpp [hunk 1]
```diff
@@ -284,7 +287,10 @@ connectionStatus_t Channel::connect(){
                 << m_pEndpoint->getHost() 
                 << ":" << m_pEndpoint->getPort() 
                 << "." << std::endl;
-            ret=this->connectInternal();
+            ret=this->setSocketInformation();
```
Comment (sanel) [inline contrib/native/client/src/clientlib/channel.cpp:290]: formatting (spaces around `=`)

File: contrib/native/client/src/clientlib/errmsgs.hpp [hunk 2]
```diff
@@ -59,7 +59,8 @@ namespace Drill{
 #define ERR_CONN_SSL_CN         DRILL_ERR_START+27
 #define ERR_CONN_SSL_CERTVERIFY DRILL_ERR_START+28
 #define ERR_CONN_SSL_PROTOVER   DRILL_ERR_START+29
-#define ERR_CONN_MAX            DRILL_ERR_START+29
+#define ERR_CONN_SSL_SNI        DRILL_ERR_START+30
+#define ERR_CONN_MAX            DRILL_ERR_START+30
```
Comment (sanel) [inline contrib/native/client/src/clientlib/errmsgs.hpp:63]: Should it be `DRILL_ERR_START+31`?

Comment (jduo) [reply contrib/native/client/src/clientlib/errmsgs.hpp:63]: Looking through the history, it looks like ERR_CONN_MAX is always set the value of the highest ERR_CONN code (as in it is an inclusive boundary, not exclusive boundary). So this seems like it should be DRILL_ERR_START+30

Comment (sanel) [reply contrib/native/client/src/clientlib/errmsgs.hpp:63]: In that case I'd just add comment for the last entry. Easy to overlook.

**Review by sanel — APPROVED** [review]
LGTM

**Comment by cgivre** [comment]
@jduo 
Can you please squash commits and we'll merge this.

**Comment by jduo** [comment]
> @jduo
> Can you please squash commits and we'll merge this.

Done @cgivre .

**Pull request merged**